server/src/proxy_server.cpp

```diff
@@ -168,10 +173,11 @@ int main(int argc, char *argv[]) {
         }
     }
 
-    void *arguments[3];
+    void *arguments[4];
     arguments[0] = malloc(sizeof(distribution * ));
```

Comment:
these malloc's aren't used? the arguments values get replaced in line 186, 191-193